### DISCUSSION TITLE
How to choose between the different audio tracks?

### DISCUSSION THREAD

**Discussion by JeanT** [section83/MacYTDL#83] [Q&A] [answered]
Hey there !
I try to download this video https://www.youtube.com/watch?v=ZMS_X-wdTi4 but it has 2 audio tracks (French and English). By default it downloads in English but I'd like to get the other. I didn't find how to do it..?
Thanks for your help !

**Comment by section83** [comment, marked as answer]
Hello JeanT,

It's a bit tricky but you can choose which audio:

- Open MacYTDL and click on "Settings".
- Tick the box beside "Get formats list" – you should get a long list of all the files available.
  - Unfortunately MacYTDL doesn't show the "More info" column which has the language.
  - There are seven (French) audio files to choose from.
  - Decide which format and bit rate you want (e.g. Opus 117k or mp4a 129k)
- Tick the box next to the related file (eg 251-0 or 140-0):
```
233-0 mp4   audio only        │                   m3u8  │ audio only           unknown             [fr] Français - original
234-0 mp4   audio only        │                   m3u8  │ audio only           unknown             [fr] Français - original
139-0 m4a   audio only      2 │   25.55MiB    49k https │ audio only           mp4a.40.5   49k 22k [fr] French original
249-0 webm  audio only      2 │   25.27MiB    48k https │ audio only           opus        48k 48k [fr] French original
250-0 webm  audio only      2 │   31.65MiB    60k https │ audio only           opus        60k 48k [fr] French original
140-0 m4a   audio only      2 │   67.81MiB   129k https │ audio only           mp4a.40.2  129k 44k [fr] French original
251-0 webm  audio only      2 │   61.30MiB   117k https │ audio only           opus       117k 48k [fr] French original
```

- Tick the box next to the video format you prefer (e.g. 137   mp4   1920x1080)
- Make sure the "Merge ?" box is ticked.
- Click on "Download".

If the result is not playable, you may have chosen an incompatible combination.  I recommend avoiding "Opus" audio and "webm" video which don't play well on macOS – but, go with what you know works on your Mac.

Post here if you have any problems.

Many thanks for making contact.


Garry

**Reply by JeanT** [reply]
Ok thank you ! So if I understand, as I can't get the "More info" column, should I rely on the ID of the audio? For example XXX-0 is for the default audio and XXX-1 is for the other?

Anyways, thank you for the quick response

**Comment by section83** [comment]
> should I rely on the ID of the audio? 

That's how it looks for that video but, I wouldn't rely on that with other videos.

I will try to add the "More Info" data to the dialog so that users can see things like language and technical descriptors (e.g. "mp4_dash).  Should be able to get that change into the next release.

Cheers.
